### DISCUSSION TITLE
Export Surface Receiver Results

### DISCUSSION THREAD

**Discussion by DiegoWdaSilva** [Ifsttar/I-Simpa#357] [Q&A]
I have set a surface receiver in a surface from my model and I want to get the Sound Intensity in that wall.

I can visualize the results in the colour plot, but I want the values. Can I export it to an external file, which then I can read in Excel or other software? I couldn't find it in the GUI, but maybe there's a script for it? Basicaly I want the 1/3 Octave Bands Spectrum for all nodes in that wall.

Also is there a way to export multiple spectrum data for point receivers? Let's say I have 100 receivers in my model, is there a way to get all spectrum instead of opening one by one and copy-paste?

Best Regards.

**Comment by nicolas-f** [comment]
Hi,

> I have set a surface receiver in a surface from my model and I want to get the Sound Intensity in that wall.
> 
> I can visualize the results in the colour plot, but I want the values. Can I export it to an external file, which then I can read in Excel or other software? I couldn't find it in the GUI, but maybe there's a script for it? Basicaly I want the 1/3 Octave Bands Spectrum for all nodes in that wall.

There is no tool to export Contours map to other software currently

The python library to open the surface files is "libsimpa"
 
> Also is there a way to export multiple spectrum data for point receivers? Let's say I have 100 receivers in my model, is there a way to get all spectrum instead of opening one by one and copy-paste?

There is no such script currently but there is a script to merge reverberation time of all receivers here:

`SystemScript/recp_res_tool/__init__.py`

best regards

**Reply by DiegoWdaSilva** [reply]
OK, so last year I didn't had that many receivers, so I ended up getting the results one by one.

Now I have a project where I have about a 1000 receivers, so it will take quite a long time to get them all.

So I tried to run this script "SystemScript/recp_res_tool/__init__.py", my idea was to try to modify the script to see if I could get the Sound Pressure Level, however this script is not running at all. It says "Python 3.8.1 TypeError: 'zip' object is not subscriptable".

I tried to fix the script, but I keep getting errors on other lines like [AttributeError: 'float' object has no attribute 'encode'].

Maybe you or someone else can help me with it?

**Reply by nicolas-f** [reply]
Hello.

As this is python the best is to manipulate the object in the interactive shell.

The first step is to find the wxid of the folder that contain all the receivers. To get that you just have to right click on the folder and copy. Paste the text in a text editor:

You will get something like that
```xml
<?xml version="1.0" encoding="UTF-8"?>
<root>
  <folder name="Punctual receivers" eid="70" wxid="34" exp="1" filepath="Punctual receivers/">
    <folder name="R1" eid="70" wxid="35" filepath="R1/">
      <arrayfile name="Punctual receiver intensity" eid="73" wxid="39" filepath="Punctual receiver intensity.gabe"/>
      <arrayfile name="Acoustic parameters" eid="73" wxid="3334" filepath="Acoustic parameters.gabe"/>
      <arrayfile name="Sound level per source" eid="110" wxid="38" filepath="Sound level per source.recps"/>
      <arrayfile name="Advanced sound level" eid="81" wxid="36" filepath="Advanced sound level.gap"/>
      <arrayfile name="Sound level" eid="74" wxid="37" filepath="Sound level.recp"/>
      <arrayfile name="Schroeder curves" eid="73" wxid="3335" filepath="Schroeder curves.gabe"/>
    </folder>
  </folder>
</root>
```

Here the wxid number is 34 (there is only one receiver in this folder R1)

In the python console type the following code (without your wxid)
```python
recplist=ui.element(34).getallelementbytype(ui.element_type.ELEMENT_TYPE_REPORT_GABE_RECP)
```

You will get a list (me it is 37). In this list it is all the noise level file (wxid) of all receivers in this folder. (for me it is 37)

You can retrieve the array in this file with this command:

```python
ui.application.getdataarray(ui.element(37))
```
You can get the receiver name with this command

```python
ui.element(ui.element(37).getinfos()["parentid"]).getinfos()["label"]
```

You can get help on any object by typing this

```python
dir(ui.element) # list all methods
help(ui.element) # list methods with some description
```
